src/Azure/Azure.Quantum.Client/Authentication/CredentialFactory.cs

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+namespace Microsoft.Azure.Quantum.Authentication
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+
+    using global::Azure.Core;
+    using global::Azure.Identity;
+
+    /// <summary>
+    /// The enumeration of supported Credential Classes supported out of the box for
+    /// authentication in Azure Quantum.
+    /// NOTE: For more information
+    /// about authentication with Azure services and the different Credential types see
+    /// https://docs.microsoft.com/en-us/dotnet/api/overview/azure/identity-readme.
```

Comment:
Nit, but in general links to docs.ms should not include language codes in the URL.
```suggestion
    /// https://docs.microsoft.com/dotnet/api/overview/azure/identity-readme.
```